### PR TITLE
Fixes #3431. TextView and TextField are showing the cursor beyond their limits.

### LINKED ISSUE #3431: TextView and TextField are showing the cursor beyond their limits.
https://github.com/gui-cs/Terminal.Gui/assets/13117724/5d0a50ef-8195-4f83-a83b-fb82dee4f04b

### PR DESCRIPTION
**Pull request by BDisp**
## Fixes

- Fixes #3431

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [ ] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

File: Terminal.Gui/Views/TextField.cs [hunk 1]
```diff
@@ -1196,13 +1196,12 @@ public override void PositionCursor ()
 
         int pos = _cursorPosition - ScrollOffset + Math.Min (Frame.X, 0);
         int offB = OffSetBackground ();
-        Rectangle containerFrame = SuperView?.ViewportToScreen (SuperView.Viewport) ?? default (Rectangle);
-        Rectangle thisFrame = ViewportToScreen (Viewport);
+        var thisOffset = ViewportToScreen (new (new (col, Viewport.Y), new (Viewport.Width - col, Viewport.Height - Viewport.Y)));
+        var view = Application.Current is { } ? FindDeepestView (Application.Current, thisOffset.X, thisOffset.Y) : this;
```
Comment (tig) [inline Terminal.Gui/Views/TextField.cs:1200]: FindDeepestView is an internal API and should no be used in any View subclass.

We need to find another way of fixing this that doesn't use an internal API.  I do not want FindDeepestView to be public.

**Review by tig — CHANGES_REQUESTED** [review]

File: Terminal.Gui/Views/TextField.cs (hunk 1 above)
Comment (BDisp) [reply Terminal.Gui/Views/TextField.cs]: I created the `IsViewLocationVisibleInViewport` method to return a bool if the view location is within the viewport. Only through the `FindDeepestView` method is it possible to confirm whether the cursor is in a visible location in the viewport.

**Review by tig — APPROVED** [review]
Unit tests are failing. 

Other than whatever that is, I think this is worth merging even though we'll eventually refactor Cursor susport per `cursor.md`.

@bdisp do you agree?

**Comment by BDisp** [comment]
> Unit tests are failing.
> 
It's not failing here.

> Other than whatever that is, I think this is worth merging even though we'll eventually refactor Cursor susport per `cursor.md`.
> 
> @BDisp do you agree?

Yes I agree.

**Comment by tig** [comment]
> > Unit tests are failing.
> 
> It's not failing here.
> 
> > Other than whatever that is, I think this is worth merging even though we'll eventually refactor Cursor susport per `cursor.md`.
> > @BDisp do you agree?
> 
> Yes I agree.

Can you please merge in the cursor.md PR?

**Comment by BDisp** [comment]
> Can you please merge in the cursor.md PR?

Done.

**Pull request merged**